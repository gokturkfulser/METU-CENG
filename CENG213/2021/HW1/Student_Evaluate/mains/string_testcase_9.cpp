#include <iostream>

#include "String.h"
//#include "../String.h"

/*
 * Case 9 : String constructor; split; print.
 */
int main() {
    std::cout << "-> Creating a string with string constructor." << std::endl;

    char chr[] = "merhaba dunya";
    String str(chr);

    std::cout << "-> Printing the string." << std::endl;

    str.print(true);

    std::cout << "-> Splitting the string with space character as the separator and printing the list." << std::endl;

    LinkedList<String> words = str.split(' ');

    words.traverse();

    std::cout << "-> Splitting the string with a character as the separator and printing the list." << std::endl;

    LinkedList<String> lSplits = str.split('a');

    lSplits.traverse();

    std::cout << "-> Printing the string." << std::endl;

    str.print(true);

    return 0;
}
