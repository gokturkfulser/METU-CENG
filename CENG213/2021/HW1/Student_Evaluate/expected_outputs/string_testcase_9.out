-> Creating a string with string constructor.
-> Printing the string.
['m','e','r','h','a','b','a',' ','d','u','n','y','a']
-> Splitting the string with space character as the separator and printing the list.
merhaba
dunya
-> Splitting the string with a character as the separator and printing the list.
merh
b
 duny

-> Printing the string.
['m','e','r','h','a','b','a',' ','d','u','n','y','a']
